src/production/logic.py

```diff
@@ -173,23 +173,46 @@ def replace_galley_file(article, request, galley, uploaded_file):
         messages.add_message(request, messages.WARNING, 'No file was selected.')
 
 
-def save_galley_image(galley, request, uploaded_file, label="Galley Image", fixed=False):
+def save_galley_image(
+        galley,
+        request,
+        uploaded_file,
+        label="Galley Image",
+        fixed=False,
+        check_for_existing_images=False
+):
+    filename = uploaded_file.name
+    # Check if an image with this name already exists for this galley.
+
+    if check_for_existing_images and galley.images.filter(
+        original_filename=filename,
+    ).exists():
+        messages.add_message(
+            request,
+            messages.WARNING,
+            f'An image called {filename} already exists. Use the replace '
+            f'file function to upload a new version.',
+        )
+        return
```

Comment:
I think also changing the default behaviour to override the matched figure by filename and galley ID with a new version is a bit tidier (As opposed to having duplicate files with idential names).
I understand that the patch to `article_figure` does resolve the reported problem, however we might see the problem reoccur on other views with similar logic.